File: Android/app/src/main/java/ru/inwords/inwords/data/source/database/WordTranslationDao.java
package ru.inwords.inwords.data.source.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import io.reactivex.Single;
import ru.inwords.inwords.data.dto.WordTranslation;

@Dao
public interface WordTranslationDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Single<Long> insert(WordTranslation wordTranslation);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Single<List<Long>> insertAll(List<WordTranslation> wordTranslation);

    @Delete
    Single<Integer> deleteAll(List<WordTranslation> wordTranslation);

    @Query("DELETE FROM word_translation_table")
    Single<Integer> deleteAll();

    @Query("DELETE FROM word_translation_table WHERE server_id IN (:serverIds)")
    Single<Integer> deleteAllServerIds(List<Integer> serverIds);

    @Query("SELECT * from word_translation_table ORDER BY word_native ASC")
    Single<List<WordTranslation>> getAllWords();
}

File: Android/app/src/main/java/ru/inwords/inwords/data/source/database/WordTranslationDao.kt
package ru.inwords.inwords.data.source.database

import androidx.room.*
import io.reactivex.Observable
import io.reactivex.Single
import ru.inwords.inwords.data.dto.WordTranslation

@Dao
interface WordTranslationDao {

    @Query("SELECT * from word_translation_table ORDER BY word_native ASC")
    fun getAllWords(): Observable<List<WordTranslation>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(wordTranslation: WordTranslation): Single<Long>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(wordTranslation: List<WordTranslation>): Single<List<Long>>

    @Delete
    fun deleteAll(wordTranslation: List<WordTranslation>): Single<Int>

    @Query("DELETE FROM word_translation_table")
    fun deleteAll(): Single<Int>

    @Query("DELETE FROM word_translation_table WHERE server_id IN (:serverIds)")
    fun deleteAllServerIds(serverIds: List<Int>): Single<Int>
}
